### PR TITLE
Update HA docs and developer setup instructions. Fix chain generation

### PR DESCRIPTION
**Pull request by plorenz**

### CODE REVIEW

File: ziti/pki/store/store.go [hunk 1]
```diff
@@ -37,14 +37,15 @@ type Store interface {
 	// Returns an error if it failed to store the bundle.
 	Add(string, string, bool, []byte, []byte) error
 
-	// Chain concats an intermediate cert and a newly signed certificate bundle and adds the chained cert to the store.
+	// Chain concats an signing cert and a newly signed certificate bundle and adds the chained cert to the store.
```
Comment (dovholuknf) [inline ziti/pki/store/store.go:40]: nit: an -> a

File: ziti/pki/store/store.go [hunk 2]
```diff
@@ -37,14 +37,15 @@ type Store interface {
...
 	//
 	// Args:
-	//  The intermediate CA name.
+	//  The signing CA name.
+	//  The destination CA name.
```
Comment (dovholuknf) [inline ziti/pki/store/store.go:44]: what's a 'destination CA'?

File: ziti/pki/store/local.go [hunk 3]
```diff
@@ -217,18 +217,23 @@ func (l *Local) writeBundle(caName, name string, isCa bool, key, cert []byte) er
...
 		}
 	}
 
-	caPath := filepath.Join(l.Root, caName, LocalCertsDir, caName+".cert")
-	caIn, err := os.Open(caPath)
+	caChainPath := filepath.Join(l.Root, caName, LocalCertsDir, caName+".chain.pem")
+	caIn, err := os.Open(caChainPath)
 	if err != nil {
-		return fmt.Errorf("failed to open CA: %v: %v", caPath, err)
+		// no chain path, likely means we're using the root as the CA, so fallback to the cert
```
Comment (dovholuknf) [inline ziti/pki/store/local.go:231]: this feels like a fail/panic situation to me? i kinda feel like this should be 'right' or not and fail without making inferences?

File: ziti/cmd/pki_create_intermediate.go [hunk 4]
```diff
@@ -123,6 +123,11 @@ func (o *PKICreateIntermediateOptions) Run() error {
 		return errors.Wrap(err, "cannot sign")
 	}
 
+	// Concat the newly-created intermediate cert with the signing cert to create an intermediate.chain.pem file
+	if err := o.Flags.PKI.Chain(signer, req); err != nil {
+		return errors.Wrap(err, "cannot sign")
```
Comment (dovholuknf) [inline ziti/cmd/pki_create_intermediate.go:128]: probably a different error here? "cannot chain"?

File: ziti/pki/store/store.go (hunk 1 above)
Comment (plorenz) [reply ziti/pki/store/store.go]: Fixed

File: ziti/pki/store/store.go (hunk 2 above)
Comment (plorenz) [reply ziti/pki/store/store.go:44]: Under which CA directory the chain should be stored. For intermediates we should store the chain with the intermediate CA, but server certs are stored in the signing CA, so we should store it in the signing CA dir.

File: ziti/pki/store/local.go (hunk 3 above)
Comment (plorenz) [reply ziti/pki/store/local.go:230]: The cases are relatively well defined. If the root is the signer, there will be no chain, if the signer is an intermediate there will be. Either way, if there's no chain, we just use the signer as the chain, which was the previous behavior.

File: ziti/cmd/pki_create_intermediate.go (hunk 4 above)
Comment (plorenz) [reply ziti/cmd/pki_create_intermediate.go]: Fixed

File: doc/ha/overview.md [hunk 5]
```diff
@@ -1,42 +1,229 @@
-# HA Setup for Development
+# OpenZiti HA
 
-**NOTE: HA is a work in progress and not yet usable for anything other than developing HA**
+This document gives a brief overview of how OpenZiti HA works and how it differs from running
+OpenZiti in non-HA mode.
```
Comment (plorenz) [inline doc/ha/overview.md:5]: Maybe add a note about certs, why to use different intermediates, spiffe ids, what's expected in the cert chain, etc

Comment (plorenz) [reply doc/ha/overview.md:5]: done

**Review by camotts — APPROVED** [review]
Looks good!

**Pull request merged**